doc/releases/changelog-0.35.0.md

```diff
@@ -45,12 +176,6 @@
   using the `@qml.defer_measurements` transform, so the above circuit would have required thousands
   of qubits to simulate.
 
-* The `default.qubit` device now supports mid-circuit measurements without using the deferred measurement
-  principle when using finite shots. The native execution mode comprises all features supported with `qml.defer_measurements`, including
-  classical control, collecting statistics, and post-selection. This PR notably introduces support for
-  post-selection, measurement value lists, and composite measurements.
-  [(#5120)](https://github.com/PennyLaneAI/pennylane/pull/5120)
-
 <h4>Work easily and efficiently with operators 🔧</h4>
 
 * Over the past few releases, PennyLane's approach to operator arithmetic has been overhauled.
```

Comment:
@isaacdevlugt @trbromley regarding this point; I feel like the positioning is a bit off here 🤔 It's focusing too much on marketing the _implementation details_ (Sum, Prod, the op_math module, etc.), but the focus of this bullet point should be: what does this allow me to do as a user of PL?